orangecontrib/text/twitter.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import tweepy
+from threading import Lock
+
+from orangecontrib.text.corpus import Corpus
+
+from Orange import data
+
+__all__ = ['Credentials', 'TwitterAPI']
+
+
+class Credentials:
+    """Twitter api credentials."""
+
+    def __init__(self, consumer_key, consumer_secret, callback=None):
+        self.consumer_key = consumer_key
+        self.consumer_secret = consumer_secret
+        self.callback = callback
+        self.auth = tweepy.OAuthHandler(self.consumer_key, self.consumer_secret,
+                                        callback=self.callback)
+
+    def check_credentials(self):
+        try:
+            self.auth.get_authorization_url()
+        except tweepy.TweepError:
+            return False
+
+        return True
+
+    def construct_api(self):
+        return tweepy.API(self.auth)
+
+    def __eq__(self, other):
+        return self.consumer_key == other.consumer_key \
+               and self.consumer_secret == other.consumer_secret
+
+
+class TwitterAPI:
+    """Fetch tweets from the Tweeter API"""
+
+    string_fields = [
+        'text', 'created_at', 'lang'
```

Comment:
Maybe, we should do this like in NY Times and have checkboxes for this different options? And then only selected fields are present in Corpus. Doing so, we could also include coordinates into the Corpus now and hopefully, someday, geomap will support this.

I agree with all @ajdapretnar's suggestions. Besides this, I would also add `coordinates` and `retweet_count`.